source/Detector/Amptek/AmptekSDD123Server.cpp

```diff
@@ -0,0 +1,522 @@
+#include "AmptekSDD123Server.h"
+
+#include <QtNetwork>
+#include <QCoreApplication>
+#include <QDebug>
+
+#include "AmptekEventDefinitions.h"
+#include "util/AMErrorMonitor.h"
+
+/* Notes on how to convert back and forth
+QString dataString = "4d4341433d3f3b";
+QString plainString = "MCAC=?;";
+qDebug() << "Normal " << dataString << plainString;
+qDebug() << "In hex " << dataString << plainString.toAscii().toHex();
+qDebug() << "In ASCII " << QString(QByteArray::fromHex(dataString.toAscii())) << plainString;
+*/
+
+AmptekSDD123Server::AmptekSDD123Server(AmptekSDD123ConfigurationMap *configurationMap, const QString &serverName, QObject *parent) :
+	QObject(parent) , currentRequestPacket_(-1, QString("NONE")), currentSyncPacket_(-1, QString("NONE")), timedOutPacket_(-1, QString("NONE"))
+{
+	configurationMap_ = configurationMap;
+	serverName_ = serverName;
+	spectrumPacketReceiver_ = 0;
+	confirmationPacketReceiver_ = 0;
+	requestIntervalTimer_ = -1;
+
+	if(configurationMap_){
+		stillReceivingResponseDatagram_ = false;
+		socketLocallyBusy_ = false;
+		detectorUnavailable_ = false;
+
+		currentRequestTime_.start();
+		requestPacketTimer_ = new QTimer(this);
+		requestPacketTimer_->setSingleShot(true);
+		syncPacketTimer_ = new QTimer(this);
+		syncPacketTimer_->setSingleShot(true);
+
+		packetIDCounter_ = 27;
+		hasReplyReady_ = false;
+
+		timeOutIDCounter_ = -27;
+
+		requestSpectrumTime_ = new QTime();
```

Comment:
Times not cleaned up.  Also, there does not appear to be a destructor to destroy them in either.